crates/bevy_ui/src/focus.rs

```diff
@@ -70,20 +79,34 @@ pub fn ui_focus_system(
     }
 
     let mouse_released =
-        mouse_button_input.just_released(MouseButton::Left) || touches_input.just_released(0);
+        mouse_button_input.get_just_released().last().is_some() || touches_input.just_released(0);
     if mouse_released {
         for (_entity, _node, _global_transform, interaction, _focus_policy) in node_query.iter_mut()
         {
             if let Some(mut interaction) = interaction {
-                if *interaction == Interaction::Clicked {
+                if matches!(*interaction, Interaction::Clicked(_)) {
                     *interaction = Interaction::None;
                 }
             }
         }
     }
 
-    let mouse_clicked =
-        mouse_button_input.just_pressed(MouseButton::Left) || touches_input.just_released(0);
+    let interacting_from = mouse_button_input
+        .get_just_pressed()
+        .last()
+        .map(|button| match button {
+            MouseButton::Left => ClickedWith::MouseLeftButton,
+            MouseButton::Middle => ClickedWith::MouseMiddleButton,
+            MouseButton::Right => ClickedWith::MouseRightButton,
+            MouseButton::Other(button) => ClickedWith::MouseOtherButton(*button),
+        })
+        .or_else(|| {
+            if touches_input.just_released(0) {
+                Some(ClickedWith::Touch)
+            } else {
+                None
+            }
```

Comment:
This could also be simplified with `then` (which is like my new favorite function btw... lol 😅)
```rust
.or_else(|| {
    touches_input.just_release(0).then(|| ClickedWith::Touch)
});
```